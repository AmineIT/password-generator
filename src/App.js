import { useState } from 'react'
import {
  Box,
  FormControl,
  InputAdornment,
  Typography,
  TextField,
  IconButton,
  FormControlLabel,
  Checkbox,
  Button,
  Slider,
  ClickAwayListener,
  Tooltip
} from '@material-ui/core'
import { FileCopy } from '@material-ui/icons'

function App() {

  const [password, setPassword] = useState('')
  const [slider, setSlider] = useState(20)
  const [lowerLetters, setLowerLetters] = useState(true)
  const [upperLetters, setUpperLetters] = useState(true)
  const [numbers, setNumbers] = useState(true)
  const [symbols, setSymbols] = useState(true)
  const [textCopied, setTextCopied] = useState(false)

  const getRandomLower = () => {
    return String.fromCharCode(Math.floor(Math.random() * 26) + 97)
  }

  const getRandomUpper = () => {
    return String.fromCharCode(Math.floor(Math.random() * 26) + 65)
  }

  const getRandomNumber = () => {
    return String.fromCharCode(Math.floor(Math.random() * 10) + 48)
  }

  const getRandomSymbol = () => {
    const symbols = "!@#$%^&*(){}/.,<>"
    return symbols[Math.floor(Math.random() * symbols.length)]
  }

  const randomFun = {
    lower: getRandomLower,
    upper: getRandomUpper,
    number: getRandomNumber,
    symbol: getRandomSymbol
  }

  const generatePassword = (slider, lower, upper, number, symbol) => {

    let generatedPassword = ''
    const typesCount = lower + upper + number + symbol
    const typesArr = [{ lower }, { upper }, { number }, { symbol }].filter(item => Object.values(item)[0])

    if (typesCount === 0) return ''

    for (let i = 0; i < slider; i += typesCount) {
      typesArr.forEach(type => {
        const funcName = Object.keys(type)[0]
        generatedPassword += randomFun[funcName]()
      })
    }

    return generatedPassword.slice(0, slider)
  }

  const handleClick = () => {
    setPassword(generatePassword(slider, lowerLetters, upperLetters, numbers, symbols))
  }

  const copyToClipboard = () => {

    if (!password) return

    setTextCopied(true)

    const textarea = document.createElement('textarea')
    textarea.value = password
    document.body.appendChild(textarea)
    textarea.select()
    document.execCommand('copy')
    textarea.remove()

  }

  const handleTooltipClose = () => {
    setTextCopied(false)
  }

  return (
    <Box
      color="white"
      display="flex"
      alignItems="center"
      justifyContent="center"
      width="100vw"
      height="100vh"
      bgcolor="#1B212C"
    >
      <Box
        padding="24px"
        borderRadius="3px"
        width="30%"
        height="auto"
        bgcolor="#171923"
      >
        <Typography variant="h5" align="center">Password Generator</Typography>

        <FormControl variant="filled" style={{ marginTop: '24px', width: '100%' }}>
          <TextField
            id="password"
            label="Password"
            variant="outlined"
            type="text"
            color="primary"
            readOnly
            value={password}
            InputLabelProps={{
              shrink: true,
            }}
            InputProps={{
              endAdornment: (
                <InputAdornment position="end">
                  <ClickAwayListener onClickAway={handleTooltipClose}>
                    <Tooltip
                      PopperProps={{
                        disablePortal: true,
                      }}
                      onClose={handleTooltipClose}
                      open={textCopied}
                      disableFocusListener
                      disableHoverListener
                      disableTouchListener
                      title="Copied!"
                    >
                      <IconButton onClick={copyToClipboard}>
                        <FileCopy style={{ color: 'white' }} />
                      </IconButton>
                    </Tooltip>
                  </ClickAwayListener>
                </InputAdornment>
              ),
            }}
          />
        </FormControl>

        <Box style={{ marginTop: '24px' }}>
          <Typography id="continuous-slider" gutterBottom>
            Password Length: {slider}
          </Typography>
          <Slider
            value={slider}
            max={50}
            min={6}
            onChange={(e, newValue) => setSlider(newValue)}
            aria-labelledby="continuous-slider" />
        </Box>

        <Box>
          <FormControlLabel
            checked={lowerLetters}
            onChange={() => setLowerLetters(!lowerLetters)}
            control={<Checkbox color="primary" />}
            label="Include lowercase letters"
            labelPlacement="end"
          />
        </Box>

        <Box>
          <FormControlLabel
            checked={upperLetters}
            onChange={() => setUpperLetters(!upperLetters)}
            control={<Checkbox color="primary" />}
            label="Include uppercase letters"
            labelPlacement="end"
          />
        </Box>

        <Box>
          <FormControlLabel
            checked={numbers}
            onChange={() => setNumbers(!numbers)}
            control={<Checkbox color="primary" />}
            label="Include numbers"
            labelPlacement="end"
          />
        </Box>

        <Box>
          <FormControlLabel
            checked={symbols}
            onChange={() => setSymbols(!symbols)}
            control={<Checkbox color="primary" />}
            label="Include symbols"
            labelPlacement="end"
          />
        </Box>

        <Button
          onClick={handleClick}
          variant="contained"
          fullWidth
          style={{ marginTop: '24px' }}>Generate</Button>
      </Box>
    </Box>
  );
}

export default App;
