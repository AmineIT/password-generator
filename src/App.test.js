import { render } from '@testing-library/react';
import App from './App';

test('render the App component', () => {
  expect(render(<App />)).toMatchSnapshot()
});
