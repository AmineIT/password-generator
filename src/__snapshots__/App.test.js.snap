// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-1"
      >
        <div
          class="MuiBox-root MuiBox-root-2"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
          >
            Password Generator
          </h5>
          <div
            class="MuiFormControl-root"
            style="margin-top: 24px; width: 100%;"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
              readonly=""
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                data-shrink="true"
                for="password"
                id="password-label"
              >
                Password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                  id="password"
                  type="text"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        style="color: white;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-7"
            style="margin-top: 24px;"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
              id="continuous-slider"
            >
              Password Length: 
              20
            </p>
            <span
              class="MuiSlider-root MuiSlider-colorPrimary"
            >
              <span
                class="MuiSlider-rail"
              />
              <span
                class="MuiSlider-track"
                style="left: 0%; width: 31.818181818181817%;"
              />
              <input
                type="hidden"
                value="20"
              />
              <span
                aria-labelledby="continuous-slider"
                aria-orientation="horizontal"
                aria-valuemax="50"
                aria-valuemin="6"
                aria-valuenow="20"
                class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
                data-index="0"
                role="slider"
                style="left: 31.818181818181817%;"
                tabindex="0"
              />
            </span>
          </div>
          <div
            class="MuiBox-root MuiBox-root-13"
          >
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-15 Mui-checked MuiIconButton-colorPrimary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input-17"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Include lowercase letters
              </span>
            </label>
          </div>
          <div
            class="MuiBox-root MuiBox-root-18"
          >
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-15 Mui-checked MuiIconButton-colorPrimary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input-17"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Include uppercase letters
              </span>
            </label>
          </div>
          <div
            class="MuiBox-root MuiBox-root-19"
          >
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-15 Mui-checked MuiIconButton-colorPrimary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input-17"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Include numbers
              </span>
            </label>
          </div>
          <div
            class="MuiBox-root MuiBox-root-20"
          >
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-15 Mui-checked MuiIconButton-colorPrimary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input-17"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Include symbols
              </span>
            </label>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-fullWidth"
            style="margin-top: 24px;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Generate
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <div
        class="MuiBox-root MuiBox-root-2"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
        >
          Password Generator
        </h5>
        <div
          class="MuiFormControl-root"
          style="margin-top: 24px; width: 100%;"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
            readonly=""
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
              data-shrink="true"
              for="password"
              id="password-label"
            >
              Password
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                id="password"
                type="text"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      style="color: white;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-7"
          style="margin-top: 24px;"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            id="continuous-slider"
          >
            Password Length: 
            20
          </p>
          <span
            class="MuiSlider-root MuiSlider-colorPrimary"
          >
            <span
              class="MuiSlider-rail"
            />
            <span
              class="MuiSlider-track"
              style="left: 0%; width: 31.818181818181817%;"
            />
            <input
              type="hidden"
              value="20"
            />
            <span
              aria-labelledby="continuous-slider"
              aria-orientation="horizontal"
              aria-valuemax="50"
              aria-valuemin="6"
              aria-valuenow="20"
              class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
              data-index="0"
              role="slider"
              style="left: 31.818181818181817%;"
              tabindex="0"
            />
          </span>
        </div>
        <div
          class="MuiBox-root MuiBox-root-13"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-15 Mui-checked MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-17"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Include lowercase letters
            </span>
          </label>
        </div>
        <div
          class="MuiBox-root MuiBox-root-18"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-15 Mui-checked MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-17"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Include uppercase letters
            </span>
          </label>
        </div>
        <div
          class="MuiBox-root MuiBox-root-19"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-15 Mui-checked MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-17"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Include numbers
            </span>
          </label>
        </div>
        <div
          class="MuiBox-root MuiBox-root-20"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-15 Mui-checked MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-17"
                  data-indeterminate="false"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Include symbols
            </span>
          </label>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-fullWidth"
          style="margin-top: 24px;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Generate
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
